internal/session/session.go

```diff
@@ -58,16 +60,23 @@ type Session struct {
 
 	// tlsConfig holds TLS information (used, for example, for STARTTLS).
 	tlsConfig *tls.Config
+
+	// imapID holds the IMAP ID extension data for this client. This is necessary, since this information may arrive
+	// before the client logs in or selects a mailbox.
+	imapID imap.ID
+
+	serverVersionInfo *internal.VersionInfo
 }
 
-func New(conn net.Conn, backend *backend.Backend, sessionID int, eventCh chan<- events.Event) *Session {
+func New(conn net.Conn, backend *backend.Backend, sessionID int, versionInfo *internal.VersionInfo, eventCh chan<- events.Event) *Session {
 	return &Session{
-		conn:      conn,
-		liner:     liner.New(conn),
-		backend:   backend,
-		caps:      []imap.Capability{imap.IMAP4rev1, imap.IDLE, imap.UNSELECT, imap.UIDPLUS, imap.MOVE},
-		sessionID: sessionID,
-		eventCh:   eventCh,
+		conn:              conn,
+		liner:             liner.New(conn),
+		backend:           backend,
+		caps:              []imap.Capability{imap.IMAP4rev1, imap.IDLE, imap.UNSELECT, imap.UIDPLUS, imap.MOVE},
+		sessionID:         sessionID,
+		eventCh:           eventCh,
+		serverVersionInfo: versionInfo,
```

Comment:
Two things: long name... could just call it "version"? And also, might be nice to include this (at least the major/minor/patch) in the greeting (see `greet()` below)

Reply:
Done